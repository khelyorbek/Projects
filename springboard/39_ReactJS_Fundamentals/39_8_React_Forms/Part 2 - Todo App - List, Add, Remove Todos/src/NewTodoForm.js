import React, { useState } from 'react';

const NewTodoForm = ({ addTodo }) => {
    const INITIAL_STATE = {
        todoTitle: ''
    }

    // using state to keep track of user's interaction with the forms
    const [formData, setFormData] = useState(INITIAL_STATE);

    // custom function to keep track of typed changed from the forms
    const handleChange = (e) => {
        const { name, value } = e.target

        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }

    // custom function to add the data into parent component's state
    const handleSubmit = (e) => {
        e.preventDefault();
        addTodo({...formData});
        // console.log(formData)
        setFormData(INITIAL_STATE);
    }    

    return (
    <>
        <form onSubmit={handleSubmit}>
            <label htmlFor="todoTitle">New task name:</label>
            <input type="text" id="todoTitle" name="todoTitle" placeholder="Type in the task name here" onChange={handleChange} value={formData.todoTitle}></input>
            <br></br>
            <button>Create new todo</button>
        </form>
    </>
        )
}

export default NewTodoForm;