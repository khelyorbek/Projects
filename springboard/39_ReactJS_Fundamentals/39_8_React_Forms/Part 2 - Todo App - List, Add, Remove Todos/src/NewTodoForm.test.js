import React from 'react';
import { render } from "@testing-library/react";
import NewTodoForm from './NewTodoForm';

// smoke test
it('renders without crashing', function() {
  render(<NewTodoForm />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});