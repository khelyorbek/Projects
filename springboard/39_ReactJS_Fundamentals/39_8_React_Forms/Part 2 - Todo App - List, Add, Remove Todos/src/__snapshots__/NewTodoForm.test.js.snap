// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todoTitle"
    >
      New task name:
    </label>
    <input
      id="todoTitle"
      name="todoTitle"
      placeholder="Type in the task name here"
      type="text"
      value=""
    />
    <br />
    <button>
      Create new todo
    </button>
  </form>
</DocumentFragment>
`;
